apps/sms/views/conversation/js/conversation.js

```diff
@@ -659,63 +657,41 @@ var ConversationView = {
     }
   },
 
-  handleForward: function conv_handleForward(forward) {
-    var request = MessageManager.getMessage(+forward.messageId);
+  handleActivity: function conv_handleActivity(params) {
+    var parametersPromise;
 
-    request.onsuccess = (function() {
-      Compose.fromMessage(request.result);
-
-      Recipients.View.isFocusable = true;
-      ConversationView.recipients.focus();
-    }).bind(this);
+    if (params.number) {
+      parametersPromise = Contacts.findByAddress(params.number).then(
+      (contacts) => {
+        if (!contacts || !contacts.length) {
+          throw new Error('No contacts found for %s', params.number);
+        }
 
-    request.onerror = function() {
-      console.error('Error while forwarding:', this.error.name);
-    };
-  },
+        return Object.assign(
+          Utils.basicContact(params.number, contacts), { source: 'contacts'}
+        );
+      }).catch(() => {
+        return { source: 'manual', number: params.number };
+      }).then((recipient) => {
+        this.recipients.add(recipient);
 
-  handleActivity: function conv_handleActivity(activity) {
-    /**
-     * Choose the appropriate contact resolver:
-     *  - if we have a phone number and no contact, rely on findByAddress
-     *    to get a contact matching the number;
-     *  - if we have a contact object and no phone number, just use a dummy
-     *    source that returns the contact.
-     */
-    var findByAddress = Contacts.findByAddress.bind(Contacts);
-    var number = activity.number;
-    if (activity.contact && !number) {
-      findByAddress = () => Promise.resolve(activity.contact);
-      number = activity.contact.number || activity.contact.tel[0].value;
-    }
-
-    // Add recipients and fill+focus the Compose area.
-    var recipientPromise;
-    if (activity.contact && number) {
-      recipientPromise = Utils.getContactDisplayInfo(findByAddress, number)
```

Comment:
you'll need to rebase after https://bugzilla.mozilla.org/show_bug.cgi?id=1153885 lands :)